src/core/evaluator.js

```diff
@@ -1623,8 +1632,6 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
               assert(isName(type), 'XObject should have a Name subtype');
 
               if (type.name !== 'Form') {
-                xobjsCache.key = name;
```

Comment:
Isn't the whole point here to XObject that does *not* have `/Subtype = /Form`?
Hence you need to replace this with `skipNonFormXObjs[name] = true;` here instead.